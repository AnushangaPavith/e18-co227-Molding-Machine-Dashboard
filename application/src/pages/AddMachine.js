import React from 'react';
import Boxes from '@material-ui/core/Box';
import InputField from '../components/InputField';
import Model from '../components/MachineAddModel';
import { useState, useEffect  } from 'react';
import axios from '../auth/axios';

const ADD_MACHINE_URL = '/api/init';

function AddNewMachine() {
    const [machineId, setMachineId]=useState("");
    const [moldID, setMoldID]=useState("");
    const [moldShots, setMoldShots] = useState(null);
    const [monaNo, setMonaNo] = useState("");
    const [material, setMaterial] = useState("");
    const [moldMaker, setMoldMaker] = useState("");
    const [moldsData, setMoldsData] = useState([]);
    const [openModel, setopenModel] = useState(false);

    // Fetch mold list from db
    useEffect(() => {
        fetch('http://localhost:3001/api/molds')
        .then(results => {
            return results.json()
        })
        .then(jsonData => {
            setMoldsData(jsonData) 
        });
    }, []);

    // Create an array to select mold from available list
    const menuItems = moldsData.map(mold => <option value={mold.moldID}>{mold.moldID}</option>);

    // When a mold is selected, the Mold data will be automatically filled
    const twoFunctions = (e) => {
        setMoldID(e);

        // Find selected mold in molds list and assign values to relevant states
        let selectedMold = moldsData.find(mold => mold.moldID === e);
        if(selectedMold !== undefined) {
            setMonaNo(selectedMold.monaNumber);
            setMaterial(selectedMold.material);
            setMoldMaker(selectedMold.moldMaker);
        }
    }

    // Submit event
    const handleSubmit = async (e) => {
        e.preventDefault();

        try {
            const response = await axios.post(ADD_MACHINE_URL, 
                JSON.stringify(
                    {
                        machineID: machineId, 
                        moldID: moldID, 
                        moldShots: moldShots, 
                        monaNumber: monaNo, 
                        material: material,
                        moldMaker: moldMaker
                    }
                ), 
                {
                    headers: { 'Content-Type' : 'application/json' }
                }
            );
            
            // When the request is success
            if(response?.data?.status === 201) {
                console.log(response?.data?.message);

                // Reset all the fields
                setMachineId('');
                setMoldID('');
                setMoldShots(null);
                setMonaNo('');
                setMaterial('');
                setMoldMaker('');
                setopenModel(true);

                // Add popup message
            }

        } catch (error) {
            if(error?.response?.data?.status === 400){
                // Almost all the errors goes here
                console.log(error?.response?.data?.message);
            }
            else {
                console.log(error?.response?.data?.message);
            }

            // Add popup for above cases
        }

    }

    return (
        <div className='new-machine-container'>
            <Boxes className='new-machine-box'>
                <div>
                    <form onSubmit={handleSubmit} >
                        <label for="Machine_ID">Machine ID</label> <br/>
                        <InputField 
                        type='text' 
                        id="Machine_ID" 
                        onChange={setMachineId}
                        value={machineId} 
                        placeholder='eg: D01'/>
                        
                        <br/>

                        <label for="Mold_ID">Mold ID</label><br/>
                        <input 
                        list='moldList' 
                        id="Mold_ID" 
                        className='input' 
                        onChange={(e) => twoFunctions(e.target.value)} 
                        autoComplete = 'off'
                        value={moldID}
                        placeholder='eg: m001'>
                        </input>
                        <datalist id="moldList">
                            {menuItems}
                        </datalist>
                        
                        <br/>
                        <br/>

                        <label for="Mold_Shots">Mold Shots</label><br/>
                        <InputField 
                        type='number' 
                        id="Mold_Shots" 
                        onChange={setMoldShots} 
                        value={moldShots} 
                        placeholder='Mold Shots'/>

                        <br/>

                        <label for="Mona_Number">Mona Number</label><br/>
                        <InputField 
                        type='text' 
                        id="Mona_Number" 
                        onChange={setMonaNo} 
                        value={monaNo} 
                        placeholder='eg: mona001'/>
                        
                        <br/>

                        <label for="Material">Material</label><br/>
                        <InputField 
                        type='text' 
                        id="Material" 
                        onChange={setMaterial} 
                        value={material} 
                        placeholder='Material'/>
                        
                        <br/>

                        <label for="Mold_Maker">Mold Maker</label><br/>
                        <InputField 
                        type='text' 
                        id="Mold_Maker" 
                        onChange={setMoldMaker} 
                        value={moldMaker} 
                        placeholder='Mold Maker'/>
                        
                        <br/>
                        
                        {/* <div className='btn'> */}
                        {/* <input type="submit" value="ADD"/> */}
                        {/* </div> */}

                        <button className='add-btn-div'>
                            Add
                        </button> 
                        
                    </form>
                </div>
            </Boxes>

            {openModel && (<Model closeModel={() => setopenModel(false) }/>)}  

        </div>
        
    );
}

export default AddNewMachine;