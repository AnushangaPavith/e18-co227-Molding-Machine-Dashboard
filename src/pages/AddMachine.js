import React from 'react';

function MachineDetailBox() {
    return ( 
        <div>

        </div>
    );
}

export default MachineDetailBox;