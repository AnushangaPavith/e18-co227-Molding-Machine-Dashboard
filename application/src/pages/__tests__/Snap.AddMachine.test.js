import { render, cleanup } from "@testing-library/react";
import AddMachine from "../AddMachine";

afterEach(() => { cleanup(); })

test("render and take snapshot of add machine page", async () => {
    const { asFragment } = render(<AddMachine />);

    expect(asFragment()).toMatchSnapshot();
})
