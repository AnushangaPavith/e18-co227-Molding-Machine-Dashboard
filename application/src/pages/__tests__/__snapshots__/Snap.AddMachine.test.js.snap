// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render and take snapshot of add machine page 1`] = `
<DocumentFragment>
  <div
    class="new-machine-container"
    data-testid="add-machine"
  >
    <div
      class="MuiBox-root MuiBox-root-1 new-machine-box"
    >
      <div>
        <form>
          <label
            for="Machine_ID"
          >
            Machine ID
          </label>
           
          <br />
          <div
            class="inputField"
          >
            <input
              class="input"
              placeholder="eg: D01"
              type="text"
              value=""
            />
          </div>
          <br />
          <label
            for="Mold_ID"
          >
            Mold ID
          </label>
          <br />
          <input
            autocomplete="off"
            class="input"
            id="Mold_ID"
            list="moldList"
            placeholder="eg: m001"
            value=""
          />
          <datalist
            id="moldList"
          />
          <br />
          <br />
          <label
            for="Mold_Shots"
          >
            Mold Shots
          </label>
          <br />
          <div
            class="inputField"
          >
            <input
              class="input"
              placeholder="Mold Shots"
              type="number"
              value=""
            />
          </div>
          <br />
          <label
            for="Mona_Number"
          >
            Mona Number
          </label>
          <br />
          <div
            class="inputField"
          >
            <input
              class="input"
              placeholder="eg: mona001"
              type="text"
              value=""
            />
          </div>
          <br />
          <label
            for="Material"
          >
            Material
          </label>
          <br />
          <div
            class="inputField"
          >
            <input
              class="input"
              placeholder="Material"
              type="text"
              value=""
            />
          </div>
          <br />
          <label
            for="Mold_Maker"
          >
            Mold Maker
          </label>
          <br />
          <div
            class="inputField"
          >
            <input
              class="input"
              placeholder="Mold Maker"
              type="text"
              value=""
            />
          </div>
          <br />
          <button
            class="add-btn-div"
          >
            Add
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
